src/http2_stream.erl

```diff
@@ -546,3 +564,21 @@ rst_stream_(ErrorCode,
                       RstStream}),
     sock:send(Socket, RstStreamBin),
     ok.
+
+check_content_length(Stream) ->
+    ContentLength =
+        proplists:get_value(<<"content-length">>,
+                            Stream#stream_state.request_headers),
+
+    case ContentLength of
```

Comment:
ok, that makes total sense, but this PR is just about request headers, and this is about response headers. 

The idea is that a HEAD request should return all the headers, exactly as if it were a GET request, just with no body.  In effect saying, if you had made a GET request, you'd have gotten this much body.